pandas/core/generic.py

```diff
@@ -2963,15 +2963,54 @@ def _update_inplace(self, result, verify_is_copy=True):
 
     def add_prefix(self, prefix):
         """
-        Concatenate prefix string with panel items names.
+        Prefix labels with string `prefix`.
+
+        For Series, the row labels are prefixed.
+        For DataFrame, the column labels are prefixed.
 
         Parameters
         ----------
-        prefix : string
+        prefix : str
+            The string to add before each label.
 
         Returns
         -------
-        with_prefix : type of caller
+        Series or DataFrame
+            Original Series or DataFrame with updated labels.
```

Comment:
This make it sound a *bit* like the object is modified inplace. Could you instead say

```
Series or DataFrame
    New Series or DataFrame with updated labels.
```

Reply:
Actually is modified inplace, should it be specified?

Reply:
Hmm I don't think so

```python
In [6]: s = pd.Series(1, ['a', 'b'])

In [7]: s
Out[7]:
a    1
b    1
dtype: int64

In [8]: s.add_prefix('foo_')
Out[8]:
foo_a    1
foo_b    1
dtype: int64

In [9]: s
Out[9]:
a    1
b    1
dtype: int64
```

`s` is unmodified.

Reply:
Sorry, I meant it's unmodified! I agree is **not** modified inplace.